include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -529,7 +529,6 @@ def TT_ReduceReturnOp: TT_Op<"reduce.return",
 def TT_ScanOp: TT_Op<"scan",
                        [Pure,
                         SameOperandsAndResultEncoding,
-                        SameOperandsAndResultElementType,
```

Comment:
Isn't there a constraint on the Operand/results matching though? Seems like this is lost here?

Reply:
We do that check here
https://github.com/openai/triton/blob/f03aae52d176615c4101176571595e77f8a0143f/lib/Dialect/Triton/IR/Ops.cpp#L597-L605
and same above for the first and second set of args.

Reply:
ah yes we are missing a `SameOperandsAndResultShape`

Reply:
Thanks! I didn't track down this verifier.
However I believe what you're pointing at is checking the matching between the region argument and the terminator operands?

I am not locating where the result of the parent ops are checked?

Reply:
actually yes each operand and result types must match individually. I guess we need a custom verifier for that? @joker-eph ?

Reply:
Can't we do it within `ScanOp::verify()`?

Reply:
ah yes as you pointed it is already done, my bad I got confused. I think it is all good

Reply:
A custom check in ScanOp::verify seems right to me?

Reply:
No, I think Mehdi has a point, albeit a bit technical one. I think he says we should be checking what we are checking on the region in the code I pointed at above. We should have the same check in `ScanOp::verify()` just to be sure. We won't be able to write a test for this as the code is already correct, but yeah, better to have it there not to violate this in the future.

Let me send a short PR

Reply:
We can write lit test to check all the verifiers :)